File: kaeltebus-backend/src/Domain/Models/Volunteer.cs
namespace kaeltebus_backend.Models;

public class Volunteer : BaseEntity
{
    public string Firstname { get; set; } = "";
    public string Lastname { get; set; } = "";
    public string Fullname { get => !String.IsNullOrEmpty(Fullname) || !String.IsNullOrEmpty(Lastname) ? $"{Firstname} {Lastname}".Trim() : ""; }
    public Gender Gender { get; set; }
    public bool IsDriver { get; set; }
}

File: kaeltebus-frontend/src/common/app/volunteer.ts
import { useCrudHook } from "../utils";
import { Gender } from "./gender";

export type Volunteer = {
  id: number;
  firstname?: string;
  lastname?: string;
  fullname?: string;
  gender?: Gender;
  isDriver?: boolean;
  remarks?: string;
};

export const useVolunteers = () =>
  useCrudHook<Volunteer, Omit<Volunteer, "id">>("volunteers");
